docs/swupd_signature_verification.md

```diff
@@ -0,0 +1,113 @@
+
+# Sign the MoM file
+openssl smime -sign -binary -in Manifest.MoM -signer Swupd_Root.pem -inkey root.key -outform DER -out Manifest.MoM.sig
+
+# Check if the signature is correct
+openssl smime -verify -in Manifest.MoM.sig -inform DER -content Manifest.MoM -purpose any -CAfile Swupd_Root.pem
+```
+
+### 2 - Using a chained certificate with a self-signed root certificate
+
+```
+# Generate a self-signed root certificate
+openssl genrsa -out root.key 4096
+openssl req -days 1 -new -x509 -key root.key -out Swupd_Root.pem -subj "/C=US/ST=Oregon/L=Portland/O=Company Name/OU=Org/CN=localhost"
+
+# Generate an intermediate key and sign it using the root certificate
+openssl genrsa -out intermediate.key 4096
+openssl req -new -key intermediate.key -out intermediate.csr -subj "/C=US/ST=Oregon/L=Portland/O=Company Name/OU=Org/CN=intermediate"
+openssl x509 -days 1 -req -in intermediate.csr -CA Swupd_Root.pem -CAkey root.key -set_serial 0x1 -out intermediate.pem
+
+# Sign the MoM file using the intermediate key
+openssl smime -sign -binary -in Manifest.MoM -signer intermediate.pem -inkey intermediate.key -out Manifest.MoM.sig -outform DER
+
+# Check if the signature is correct using the root key
+openssl smime -verify -in Manifest.MoM.sig -inform DER -content Manifest.MoM -purpose any -CAfile Swupd_Root.pem
+
+# Note that in this example the key used to sign the updates is the intermediate key and the certificate used by clients to validate the signature is the Swupd_Root.pem
+```
+
+### 2b - Rotating the intermediate key
+
+One advantage on using the solution 2 is that it’s possible to rotate the intermediate key if necessary. Instructions on how to perform this operation are following.
+
+```
+# Generate and sign the new intermediate key
+openssl genrsa -out new_intermediate.key 4096
+openssl req -new -key new_intermediate.key -out new_intermediate.csr -subj "/C=US/ST=Oregon/L=Portland/O=Company Name/OU=Org/CN=new_intermediate"
+openssl x509 -days 1 -req -in new_intermediate.csr -CA Swupd_Root.pem -CAkey root.key -set_serial 0x1 -out new_intermediate.pem
+
+# Sign the MoM file using the new key
+openssl smime -sign -binary -in Manifest.MoM -signer new_intermediate.pem -inkey new_intermediate.key -out Manifest.MoM.sig -outform DER
+
+# Check if the signature is correct using the same root key
+openssl smime -verify -in Manifest.MoM.sig -inform DER -content Manifest.MoM -purpose any -CAfile Swupd_Root.pem
+```
+
+## Certificate Revocation
+
+### Certificate Revocation List
+
+Usage of Certificate Revocation list is implemented in src/signature.c, but isn’t currently used on swupd. To enable that calls to signature_init() should set the crl parameter pointing to a file with the Certificate Revocation List.
+
+### OCSP (Online Certificate Status Protocol)
+
+OCSP isn’t yet supported on swupd. If a certificate with OCSP set and flagged as critical is used swupd will abort with a warning saying that OCSP isn’t supported.
```

Comment:
If a certificate with OCSP set and flagged as critical is used swupd will abort  ->  If a certificate with OCSP set and flagged as critical is used, swupd will abort